laiaaa0/assignment3/q1.cpp

```diff
@@ -0,0 +1,51 @@
+
+#include "TestingFunctions.cpp"
+
+int directions_x []= {1,-1,0,0,1,-1,-1,1};
+int directions_y []= {0,0,1,-1,1,1,-1,-1};
+
+bool IsInGrid (int i, int j,const std::vector<std::vector<char> > &grid){
+    if (i>=0 && i<grid.size()){
+      if (j>=0 && j<grid[i].size()) return true;
+    }
+    return false;
+}
+
+void FindAllWordsRecursive (int x, int y, Dictionary * d, std::vector<std::vector<char> > &grid, std::vector<std::vector<bool> > &visited,  std::string current_word,std::vector<std::string> &found_words){
```

Comment:
Similar to the "const reference" convention, a common practice is to make sure that anything that you can modify is passed as a "non-const pointer". This help make it clearer what it can and cannot be modified.